interview-questions.md

```diff
@@ -7,50 +7,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on Rails is a framework built inside a ruby gem that provides a scalable framework with base functionality built into the gem itsself.
 
-  Researched answer:
+  Researched answer: Rails is a server side web app framework  written in ruby. It provides default structures for a database, webservice, and web pages. It facilitates the use of web standards like JSON and XML for data transfer and HTML, CSS and JS for UI. It emphasizes use of convention over configuration, DRY, and the active record pattern. 
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is one that uses unique keys for each of its values and uses that relationship to call on the value. probably
 
-  Researched answer:
+  Researched answer: Each row in the table is a record with a unique key. The comlumns of the table hold attribute of the data and each record usually has a value for each attribute making it easy to establish the attributes relationships. 
+
+
+  A - atomicity defines all the elements that make up a complete database transaction.
+  C - consistency defines the rules for maintaining data points in a correct state after a transaction 
+  I - Isolation keeps the effect of a transaction invisible to others until it is completed, to avoid confusion
+  D - Durability ensures that data changes become permanent once the transaction is committed
+
+  There are non relational databases, NoSQL.
+  Document data store
+  column-oriented database
+  key-value store and graph db
+  and graph db 
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are the unique keys that are in the database that the value is in, and foreign keys are those keys stored in other databases. They are important becuase they are the location of the values within the database.
 
-  Researched answer:
+  Researched answer: A primary key is used as a unique identifier for a column or groups of columns and helps to quickly parse data within the table. Without primary keys relational databases wouldn't work. 
 
 
 
 ## Looking Ahead: Terms for Next Week
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: REpresentational State Transfer provides a consistent pattern to use in structuring routes. It provides mapping from HTTP verbs (get, post, delete, patch) to controller CRUD actions. a RESTful route uses the HTTP verb and the URL.
+
+2. JSON: JS Object Notation is a data interchange format. It is easy to read and write, and parse and generate.
+
+I had a horrible weekend, really week and a half but the weekend was the worst so this is as far as I could get. :cry:
```

Comment:
Everything you contributed to this assessment looks fantastic. I'm sorry you had a rough week. I hope you feel better.